src/Microsoft.NetCore.Analyzers/Core/Security/DoNotInstallRootCert.cs

```diff
@@ -0,0 +1,223 @@
+
+                case PropertySetAbstractValueKind.MaybeFlagged:
+                    return HazardousUsageEvaluationResult.MaybeFlagged;
+
+                default:
+                    return HazardousUsageEvaluationResult.Unflagged;
+            }
+        }
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            HazardousUsageEvaluatorCollection hazardousUsageEvaluators = new HazardousUsageEvaluatorCollection(
+                new HazardousUsageEvaluator("Add", HazardousUsageCallback));
+
+            context.RegisterCompilationStartAction(
+                (CompilationStartAnalysisContext compilationStartAnalysisContext) =>
+                {
+                    var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(compilationStartAnalysisContext.Compilation);
+
+                    if (!wellKnownTypeProvider.TryGetTypeByMetadataName(WellKnownTypeNames.SystemSecurityCryptographyX509CertificatesX509Store, out var x509TypeSymbol))
+                    {
+                        return;
+                    }
+
+                    if (!wellKnownTypeProvider.TryGetTypeByMetadataName(WellKnownTypeNames.SystemSecurityCryptographyX509CertificatesStoreName, out var storeNameTypeSymbol))
+                    {
+                        return;
+                    }
+
+                    // If X509Store is initialized with Root store, then that instance is flagged.
+                    var constructorMapper = new ConstructorMapper(
+                        (IMethodSymbol constructorMethod, IReadOnlyList<ValueContentAbstractValue> argumentValueContentAbstractValues,
+                        IReadOnlyList<PointsToAbstractValue> argumentPointsToAbstractValues) =>
+                        {
+                            var kind = PropertySetAbstractValueKind.Unflagged;
+
+                            if (constructorMethod.Parameters.Length > 0)
+                            {
+                                var valueContent = argumentValueContentAbstractValues[0].LiteralValues;
+
+                                if (constructorMethod.Parameters[0].Type.Equals(storeNameTypeSymbol) &&
+                                    valueContent.Contains(6) ||
+                                    constructorMethod.Parameters[0].Type.SpecialType == SpecialType.System_String &&
+                                    valueContent.Any(s => string.Equals(s.ToString(), "root", StringComparison.OrdinalIgnoreCase)))
+                                {
+                                    kind = valueContent.Count == 1 ? PropertySetAbstractValueKind.Flagged : PropertySetAbstractValueKind.MaybeFlagged;
```

Comment:
>valueContent.Count == 1  [](start = 43, length = 24)

If there were three values, 6, "root", and "ROOT", you'd want to treat that as flagged.

Reply:
Yes, I think Contains one of these value is good. I won't flag this situation as MaybeFlagged.

Reply:
I liked your previous approach of:
- All bad values => Flagged
- Some, but not all, bad value => MaybeFlagged

I'll change other places in the code to do the same.

Reply:
Agree. But the code will be long.

Reply:
I'll write a helper method to do this.  Your code is fine for now.